x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_time_range_url_state.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fold } from 'fp-ts/lib/Either';
+import { constant, identity } from 'fp-ts/lib/function';
+import { pipe } from 'fp-ts/lib/pipeable';
+import * as rt from 'io-ts';
+import { useUrlState } from '../../../../utils/use_url_state';
+import {
+  useKibanaTimefilterTime,
+  useSyncKibanaTimeFilterTime,
+} from '../../../../hooks/use_kibana_timefilter_time';
+
+export const stringTimeRangeRT = rt.type({
+  startTime: rt.string,
+  endTime: rt.string,
+});
+export type StringTimeRange = rt.TypeOf<typeof stringTimeRangeRT>;
+
+const urlTimeRangeRT = rt.union([stringTimeRangeRT, rt.undefined]);
+
+const TIME_RANGE_URL_STATE_KEY = 'timeRange';
```

Comment:
For consistency, we might want to have a constant for the state key for the filter and query hooks too

Reply:
Thank you for spotting that, I added the constants inside the other hooks.